pkg/operator/boundsatokensignercontroller/controller.go

```diff
@@ -0,0 +1,303 @@
+package boundsatokensignercontroller
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"k8s.io/klog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	workQueueKey = "key"
+
+	operatorNamespace = operatorclient.OperatorNamespace
+	targetNamespace   = operatorclient.TargetNamespace
+
+	keySize                  = 2048
+	NextSigningKeySecretName = "next-service-account-signing-key"
+	SigningKeySecretName     = "service-account-signing-key"
+	PrivateKeyKey            = "service-account.key"
+	PublicKeyKey             = "service-account.pub"
+
+	// When creating a new keypair, the following annotation will be
+	// added to the operator's signing key secret and set to now + the
+	// ready interval. Only once that interval has elapsed will the
+	// contents of the operator secret be promoted to the operand
+	// secret for use by the apiserver. This interval is intended to
+	// ensure that apiservers have a chance to update to the latest
+	// public key configmap that can validate tokens issued by the new
+	// private key. If the secret was promoted immediately, and new
+	// tokens were issued by its private key, apiserver instances that
+	// did not have the latest public key would not be able to
+	// validate those new tokens.
+	TokenReadyAnnotation = "kube-apiserver.openshift.io/ready-to-use"
+	readyInterval        = 5 * time.Minute
```

Comment:
You have the ability to inspect the revisions to figure out which ones contains the key in question.  You can then inspect the levels on the nodes (the kubeapiservers.operator.openshift.io) to know if the nodes actually have the levels required.  No need for time.

Reply:
Done